src/tests/queue-test/queue-test.c

```diff
@@ -172,9 +224,25 @@ void QueueTimeoutSetup(void)
     status = OS_TimerSet(timer_id, timer_start, timer_interval);
     UtAssert_True(status == OS_SUCCESS, "Timer 1 set Rc=%d", (int)status);
 
-    /* allow some time for task to run and accrue queue timeouts */
-    while (timer_counter < 100)
+    /* Put 10 messages onto the que with some time inbetween to not overfill the que*/
+    for (i = 0; i < 10; i++)
     {
-        OS_TaskDelay(100);
+        OS_TaskDelay(200);
```

Comment:
While I understand the goal of not wanting to overfill the queue - it also misses an important objective of confirming that the queue will actually queue/store data up to the requested depth limit.  This is part of the function that we really should test - that we can write multiple messages (e.g. at least 2) when the receiver is _not_ waiting to receive them - then when the receiver does call `OS_QueueGet()` later - it still gets them in the original order.  

In this test we should be guaranteed to be able to write up to `MSGQ_DEPTH` and not lose any data.  In fact this can be done in a single thread (write into queue up to `MSGQ_DEPTH`, then read queue and confirm all data came back in same order with nothing missing).

Reply:
While it is not clear from the code this is testing that the que stores data. It sends a message every 200 milliseconds and the que doesn't start reading until after 1 second. That way when the que starts there are 5 messages waiting for it and then 5 more come spread out over 200 milliseconds each. I did it this way to test both messages waiting for the que to open and the que waiting for more messages to recieve. 

For the sake of clarity I could split that into two seperate test if we think that would be better.

Reply:
changed it not wait in the first few puts and added better comments to better express that some messages are waiting in the que.